cmd/dependabot/internal/cmd/update.go

```diff
@@ -156,6 +85,86 @@ var updateCmd = &cobra.Command{
 	},
 }
 
+func extractInput(cmd *cobra.Command) (*model.Input, []byte, error) {
+	var input *model.Input
+	var inputRaw []byte
+
+	if file != "" {
+		if len(cmd.Flags().Args()) > 0 {
+			return nil, nil, errors.New("cannot use file and arguments together")
+		}
+		var err error
+		input, inputRaw, err = readInputFile(file)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	if len(cmd.Flags().Args()) > 0 {
```

Comment:
I'd expect this to drop into a dedicated function for parsing the input flags...